packages/flutter/lib/src/material/dialog.dart

```diff
@@ -1129,3 +1140,71 @@ double _paddingScaleFactor(double textScaleFactor) {
   // a non-scaled padding of 24 will produce a padding between 24 and 8.
   return lerpDouble(1.0, 1.0 / 3.0, clampedTextScaleFactor - 1.0)!;
 }
+
+// Generate a DialogTheme that represents the M2 default values.
+// This was generated by hand from the previous hand coded defaults
+// for M2. It uses get method overrides instead of properties to
+// avoid computing values that we may not need upfront.
+class _M2Defaults extends DialogTheme {
+  _M2Defaults(this.context)
+    : _textTheme = Theme.of(context).textTheme;
+
+  final BuildContext context;
+  final TextTheme _textTheme;
+
+  @override
+  AlignmentGeometry? get alignment =>  Alignment.center;
+
+  @override
+  Color? get backgroundColor => Theme.of(context).dialogBackgroundColor;
+
+  @override
+  double? get elevation => 24.0;
+
+  @override
+  ShapeBorder? get shape => const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(4.0)));
+
+  @override
+  TextStyle? get titleTextStyle => _textTheme.headline6;
+
+  @override
+  TextStyle? get contentTextStyle => _textTheme.subtitle1;
+}
+
+// BEGIN GENERATED TOKEN PROPERTIES
+
+// Generated code to the end of this file. Do not edit by hand.
+// These defaults are generated from the Material Design Token
+// database by the script dev/tools/gen_defaults/bin/gen_defaults.dart.
+
+// Generated version v0_81
+class _M3Defaults extends DialogTheme {
+  _M3Defaults(this.context)
+    : _colors = Theme.of(context).colorScheme,
+      _textTheme = Theme.of(context).textTheme;
+
+  final BuildContext context;
+  final ColorScheme _colors;
+  final TextTheme _textTheme;
+
+  // Not derived from tokens.
+  @override
+  AlignmentGeometry? get alignment =>  Alignment.center;
+
+  @override
+  Color? get backgroundColor => ElevationOverlay.colorWithOverlay(_colors.surface, _colors.primary, 6.0);
+
+  @override
+  double? get elevation => 6.0;
+
+  @override
+  ShapeBorder? get shape => const RoundedRectangleBorder(borderRadius: BorderRadius.only(topLeft: Radius.circular(28.0), topRight: Radius.circular(28.0), bottomLeft: Radius.circular(28.0), bottomRight: Radius.circular(28.0)));
```

Comment:
I think ``BorderRadius.only(...)`` could be simplified to ``BorderRadius.all(Radius.circular(28.0))``